test/e2e/network/service.go

```diff
@@ -2820,21 +2821,28 @@ var _ = common.SIGDescribe("Services", func() {
 		nodeIPs := e2enode.GetAddresses(&node0, v1.NodeInternalIP)
 		healthCheckNodePortAddr := net.JoinHostPort(nodeIPs[0], strconv.Itoa(int(svc.Spec.HealthCheckNodePort)))
 		// validate that the health check node port from kube-proxy returns 200 when there are ready endpoints
-		err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
-			cmd := fmt.Sprintf(`curl -s -o /dev/null -w "%%{http_code}" --max-time 5 http://%s/healthz`, healthCheckNodePortAddr)
-			out, err := e2eoutput.RunHostCmd(pausePod0.Namespace, pausePod0.Name, cmd)
-			if err != nil {
-				framework.Logf("unexpected error trying to connect to nodeport %s : %v", healthCheckNodePortAddr, err)
-				return false, nil
-			}
+		err = wait.PollUntilContextTimeout(ctx,
+			time.Second,
+			time.Minute,
+			true,
+			func(ctx context.Context) (bool, error) {
+				cmd := fmt.Sprintf(`curl -s -o /dev/null -w "%%{http_code}" --max-time 5 http://%s/healthz`,
+					healthCheckNodePortAddr)
+				out, err := e2eoutput.RunHostCmd(pausePod0.Namespace, pausePod0.Name, cmd)
+				if err != nil {
+					framework.Logf("unexpected error trying to connect to nodeport %s : %v",
+						healthCheckNodePortAddr,
+						err)
+					return false, nil
+				}
```

Comment:
also unrelated, keep it as is

Reply:
Both changes have been removed, but iirc they were made because the job `pull-kubernetes-verify` failed the linting otherwise